base/src/main/java/org/aya/resolve/visitor/ExprResolver.java

```diff
@@ -84,14 +87,21 @@ public void enterBody() {
     return new ExprResolver(ctx, RESTRICTIVE, allowedGeneralizes, reference, MutableStack.create(), this::addReference);
   }
 
+  private @Nullable Stmt.Accessibility accessibility() {
+    return switch (where.peek()) {
+      case Where.Body body -> null;
+      case Where.Head head -> head.accessibility();
+    };
+  }
+
```

Comment:
Why need this? The `accessibility` is the accessibility of the currently being-resolved definition. It has no effect on `checkAccessibility`.

![image](https://user-images.githubusercontent.com/9902643/214510351-4192afea-e48f-453f-8f0e-d754d182685c.png)

Reply:
I even believe the check of accessibility should be done in `ExprResolver` rather than here.

Reply:
> Why need this? The `accessibility` is the accessibility of the currently being-resolved definition. It has no effect on `checkAccessibility`.

`Context#get` -> `Context#getUnqualified` -> `Context#getUnqualifiedLocal` -> `Context#getUnqualifiedLocalMaybe` -> `Context#checkAccessibility`

Reply:
This looks scary and it needs some hacking in PhysicalModuleContext.java (the `permits`). What will go wrong if we put the accessibility check just in `ExprResolver` after `ctx.get(name).data()`

Reply:
The impl of `checkAccessibility` always gets called after every unqualified resolve, and it says "public definitions cannot refer to private ones". There are problems:
- What should I do when I just want to hide the detail to other modules?
- Not all places need to check accessibility, for example, as you said: `private data N | zero | suc N`